### PR TITLE
Staking Typescript SDK and page fixes

### PR DESCRIPTION
**Pull request by NightTrek**

### CODE REVIEW

File: package-lock.json
Comment (Chazzzzzzz) [inline package-lock.json:1]: do we need this?

File: src/components/forms/NetworkSelect.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import { useState, useRef } from "react";
+import Image from "next/image";
+import { useOnClickOutside } from "../../hooks/useOnClickOutside";
+
+export const NETWORKS = {
```
Comment (Chazzzzzzz) [inline src/components/forms/NetworkSelect.js:5]: shall we add goerli here as well?

File: src/components/modals/ApprovePM.js [hunk 2]
```diff
@@ -15,7 +15,8 @@ export default function ApprovePM(props) {
   function approve(){
     setLoading(true);
     setApproving(true)
-    handleApprove(wallet, eoa, paymasterAddress, paymentAddr, amount).then((data) => {
+    console.log(paymasterAddress, paymentAddr)
```
Comment (Chazzzzzzz) [inline src/components/modals/ApprovePM.js:18]: pls remove

File: src/components/modals/ConnectWallet.js [hunk 3]
```diff
@@ -1,259 +1,266 @@
...
-          chainId: 5,
-          rpcUrl: "https://rpc.ankr.com/eth_goerli"
-        },
-        extensions: [new OAuthExtension()],
-      })
+    useEffect(() => {
+        const initMagicAuth = async () => {
+            const magic = new Magic("pk_live_846F1095F0E1303C", {
+                network: {
+                    chainId: 5,
```
Comment (Chazzzzzzz) [inline src/components/modals/ConnectWallet.js:35]: this should probably change according to network select

File: src/components/modals/Main.js [hunk 4]
```diff
@@ -1,81 +1,104 @@
...
+                    <div className="ml-3">
+                        <div className="font-medium text-[#344054]">Stake Ethereum</div>
+                        <div className="text-[#667085]">Stake to earn rewards. Longer staking earn more rewards.</div>
+                    </div>
+                </div>
+                {/* <div className="button flex items-center text-sm mt-4 p-4" onClick={() => goToPage("/unstake")}>
+                    <Image src="/lock-closed-circular.svg" width="40" height="40" alt="" />
+                    <div className="ml-3">
+                        <div className="font-medium text-[#344054]">Unstake Ethereum</div>
+                        <div className="text-[#667085]">Unstake and claim Ethereum on Goerli.</div>
```
Comment (Chazzzzzzz) [inline src/components/modals/Main.js:91]: pls remove if not needed

File: src/contexts/funContext.js [hunk 5]
```diff
@@ -21,10 +22,18 @@ export const FunProvider = ({ children }) => {
...
       router.push('/connect');
     }
-  }, [wallet, network, router.pathname])
+  }, [wallet, network, router.pathname, router])
 
+  // useEffect(() => {
+  //   if (provider == null) return;
+  //   provider.onChainChanged((chainId) => {
+  //     setNetwork(parseInt(chainId))
+  //   })
```
Comment (Chazzzzzzz) [inline src/contexts/funContext.js:31]: pls remove if not needed

File: src/scripts/bridge.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+                        success: false,
+                        mustApprove: true,
+                        paymasterAddress,
+                        tokenAddr: paymentToken
+                    }
+                }
+            } else {
+                return {
+                    success: false,
+                    error: "Its a known bug that first transaction of a fun wallet would fail if you are covering gas using ERC20 tokens. Please try to pay gas using gasless paymaster or ETH for this transaction and try token paymaster later."
```
Comment (Chazzzzzzz) [inline src/scripts/bridge.js:86]: we should remove all of those for now given the issue is fixed

File: src/scripts/bridge.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+                chain: fromChainId,
+                apiKey,
+                gasSponsor: {
+                    sponsorAddress: "0x175C5611402815Eba550Dad16abd2ac366a63329"
+                }
+            }
+        }
+
+        try {
+            console.log(tempEnvOptions)
```
Comment (Chazzzzzzz) [inline src/scripts/bridge.js:100]: pls remove

File: src/scripts/bridge.js [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+            const receipt = await wallet.bridge(auth, defaultParamsObj, {
+                ...tempEnvOptions,
+                gasLimit: 300000
+            })
+            //Tells frontend stake was success
+            console.log("txId: ", receipt, receipt.txid)
+            const explorerUrl = receipt.txid
+                ? `https://goerli.etherscan.io/tx/${receipt.txid}`
+                : `https://goerli.etherscan.io/address/${walletAddress}#internaltx`
+            console.log(receipt)
```
Comment (Chazzzzzzz) [inline src/scripts/bridge.js:110]: pls remove

File: src/scripts/mintNFT.js [hunk 9]
```diff
@@ -1,75 +1,42 @@
...
-        return { success: false, error: "Its a known bug that first transaction of a fun wallet would fail if you are covering gas using ERC20 tokens. Please try to pay gas using gasless paymaster or ETH for this transaction and try token paymaster later." }
-      }
-    }
-    else if(paymentToken=="gasless"){
-      await configureEnvironment({
-        chain: 5,
-        apiKey,
-        gasSponsor: {
-          sponsorAddress: '0x07Ac5A221e5b3263ad0E04aBa6076B795A91aef9',
+export const handleMintNFT = async function (wallet, paymentToken, nft, auth) {
```
Comment (Chazzzzzzz) [inline src/scripts/mintNFT.js:8]: do we need to checkIfWalletIsPrefunded for bridge?

File: src/scripts/swap.js [hunk 10]
```diff
@@ -1,106 +1,91 @@
...
-      }
-    }
-    else if(paymentToken=="gasless"){
-      await configureEnvironment({
-        chain: 5,
-        apiKey,
-        gasSponsor: {
-          sponsorAddress: '0x07Ac5A221e5b3263ad0E04aBa6076B795A91aef9',
+            try {
+                // balance = await Token.getBalance(inAddr, walletAddress)
```
Comment (Chazzzzzzz) [inline src/scripts/swap.js:39]: is this intended?

File: src/scripts/wallet.js [hunk 11]
```diff
@@ -1,90 +1,231 @@
...
+        //GOERLI
+        try {
+            await handleFundWallet(addr)
+            setTimeout(() => {
+                return
+            }, 1500)
+        } catch (e) {}
+    } else if (network == 1) {
+        //MAINNET
+        try {
```
Comment (Chazzzzzzz) [inline src/scripts/wallet.js:60]: we should pop up an error saying we do not support faucet on mainnet/polygon and BSC

File: src/scripts/wallet.js [hunk 12]
```diff
@@ -1,90 +1,231 @@
...
+// check if the wallet is deployed
+// check if the wallet has enough funds and if not fund the wallet
+export const checkIfWalletIsPrefunded = async (wallet, estimatedGas, chainId, native = true) => {
+    if (!native) return { success: true }
+    try {
+        const walletAddress = await wallet.getAddress()
+        const etherBalance = await Token.getBalance("ETH", walletAddress)
+        const balance = ethers.utils.parseEther(etherBalance)
+        console.log("balance", balance.toString(), "estimatedGas", estimatedGas.toString(), balance < estimatedGas)
+        if ( estimatedGas.gt(balance)) {
```
Comment (Chazzzzzzz) [inline src/scripts/wallet.js:176]: should compare between estimatedGas * gasPrice and balance

Comment (Chazzzzzzz) [reply src/scripts/wallet.js:176]: aaron should have an API to get gasPrice across chains

File: src/utils/constants.js [hunk 13]
```diff
@@ -1,3 +1,5 @@
-const apiKey = "MYny3w7xJh6PRlRgkJ9604sHouY2MTke6lCPpSHq"
+// const apiKey = "MYny3w7xJh6PRlRgkJ9604sHouY2MTke6lCPpSHq"
```
Comment (Chazzzzzzz) [inline src/utils/constants.js:1]: pls remove

**Review by Chazzzzzzz — COMMENTED** [review]
comments

File: src/components/forms/NetworkSelect.js (hunk 1 above)
Comment (NightTrek) [reply src/components/forms/NetworkSelect.js:5]: its for bridging only. so no test nets

File: src/scripts/wallet.js (hunk 12 above)
Comment (NightTrek) [reply src/scripts/wallet.js:176]: This function determine if the user has enough Eth in their fun wallet to initialize it. It automatically  funds their wallet if there is not and returns an error

**Review by Chazzzzzzz — APPROVED** [review]

**Pull request merged**